relations_procmacro/tests/01-get-corresponding.rs

```diff
@@ -0,0 +1,63 @@
+mod test_utils;
+
+use relations::*;
+use relations_procmacro::*;
+use test_utils::*;
+use typed_index_collection::*;
+
+#[derive(GetCorresponding)]
+pub struct Model {
+    animals_to_felines: OneToMany<Animal, Feline>,
+    felines_to_cats: OneToMany<Feline, Cat>,
+}
+
+fn main() {
+    let animal = Animal {
+        id: String::from("animal_id"),
+    };
+    let feline1 = Feline {
+        id: String::from("feline_id_1"),
+        animal_id: String::from("animal_id"),
+    };
+    let feline2 = Feline {
+        id: String::from("feline_id_2"),
+        animal_id: String::from("animal_id"),
+    };
+    let cat_1 = Cat {
+        id: String::from("cat_id_1"),
+        feline_id: String::from("feline_id_1"),
+    };
+    let cat_2 = Cat {
+        id: String::from("cat_id_2"),
+        feline_id: String::from("feline_id_1"),
+    };
+    let cat_3 = Cat {
+        id: String::from("cat_id_3"),
+        feline_id: String::from("feline_id_2"),
+    };
+    let cat_4 = Cat {
+        id: String::from("cat_id_4"),
+        feline_id: String::from("feline_id_2"),
+    };
+    let animals = CollectionWithId::from(animal);
+    let felines = CollectionWithId::new(vec![feline1, feline2]).unwrap();
+    let cats = CollectionWithId::new(vec![cat_1, cat_2, cat_3, cat_4]).unwrap();
+    let model = Model {
+        animals_to_felines: OneToMany::new(&animals, &felines, "animals_to_felines").unwrap(),
+        felines_to_cats: OneToMany::new(&felines, &cats, "felines_to_cats").unwrap(),
+    };
+
+    let animal_idx = animals.get_idx("animal_id").unwrap();
+    let cat_indexes = model.get_corresponding_from_idx(animal_idx);
+    let cat_1_idx = cats.get_idx("cat_id_1").unwrap();
+    assert!(cat_indexes.contains(&cat_1_idx));
+    let cat_2_idx = cats.get_idx("cat_id_2").unwrap();
+    assert!(cat_indexes.contains(&cat_2_idx));
+    let cat_3_idx = cats.get_idx("cat_id_3").unwrap();
+    assert!(cat_indexes.contains(&cat_3_idx));
+    let cat_4_idx = cats.get_idx("cat_id_4").unwrap();
+    assert!(cat_indexes.contains(&cat_4_idx));
+
+    let animal_indexes = model.get_corresponding_from_idx(cat_1_idx);
+    assert!(animal_indexes.contains(&animal_idx));
```

Comment:
it would be good to also test `get_corresponding` with a set o Idx

Reply:
These were the existing tests that I just copied. I just modified them to not depend on `transit_model` (by introducing `Animal`, `Feline` and `Cat`). However, let me see if I can do something about it.

Reply:
Done in d36ffef887dd88bda03ab3ce0c93d16a5b4ced35. Let me know if this is what you had in mind?

Reply:
no :) I wanted you to test `model.get_corresponding(idxset)`

Reply:
Done in 26b59e494f7c37f6c7ef72c04c138e1f13da5546.